components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/utils/InboundWebsocketProcessorUtil.java

```diff
@@ -675,7 +676,8 @@ public static InboundProcessorResponseDTO authenticateToken(InboundMessageContex
         InboundProcessorResponseDTO inboundProcessorResponseDTO = new InboundProcessorResponseDTO();
         try {
             //validate token and subscriptions
-            if (!InboundWebsocketProcessorUtil.authenticateGraphQLJWTToken(inboundMessageContext)) {
+            if (inboundMessageContext.isJWTToken() && !InboundWebsocketProcessorUtil.authenticateGraphQLJWTToken(
+                    inboundMessageContext)) {
```

Comment:
shall we add a else and add a log debug stating, authentication no supported for oauth tokens.